google_guest_agent/events/events.go

```diff
@@ -48,14 +49,65 @@ type Watcher interface {
 // Manager defines the interface between events management layer and the
 // core guest agent implementation.
 type Manager struct {
```

Comment:
can we use `sync.RWMutex` here for any of the attributes? In some cases parallel reads should be fine, we can protect just the writes?

Reply:
Changed wherever it made sense. Thank you.